ENGAGEHF/Bluetooth/MeasurementManager.swift

```diff
@@ -0,0 +1,230 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import BluetoothServices
+import FirebaseAuth
+import FirebaseCore
+import FirebaseFirestore
+import Foundation
+import HealthKit
+import OSLog
+import Spezi
+@_spi(TestingSupport) import SpeziBluetooth
+import SpeziFirestore
+
+
+// Functionality:
+// - Store the user's measurement history in an array
+//      - Watch for changes in the user's blood pressure and weight measurement collections in firebase
+// - Convert an incoming measurement into a HKSample and transform to FHIR Observation
+// - Save a given measurement to Firebase
+@Observable
+class MeasurementManager: Module, EnvironmentAccessible {
+    private static var _manager: MeasurementManager?
+    static var manager: MeasurementManager {
+        guard let manager = _manager else {
+            fatalError("Accessing shared MeasurmentManager before initialized.")
+        }
+        return manager
+    }
+    
+    @ObservationIgnored @StandardActor private var standard: ENGAGEHFStandard
+    private let logger = Logger(subsystem: "ENGAGEHF", category: "MeasurementManager")
+    
+    var deviceInformation: DeviceInformationService?
+    var weightScaleParams: WeightScaleFeature?
+    var deviceName: String?
+    
+    var newMeasurement: HKQuantitySample?
```

Comment:
I am wondering if any of these properties could be private or at least `private(set)`?

Reply:
At the moment, these are directly set by the WeightScaleDevice on change of the WeightMeasurement characteristic. This is heavily reliant on the singleton implementation we've been using, as the WeightScaleDevice does not have access to the environment to import the MeasurementManager. I can make these private(set) by making a function in MeasurementManager that sets the variables internally.

Reply:
@nriedman No need to add an extra function at this point; we might to rethink the need for the singles in the future as noted in the previous comments here in combination with improvements in Spezi Bluetooth: https://github.com/StanfordBDHG/ENGAGE-HF/pull/6#discussion_r1600236080

For now, we can keep it like this.

CC: @Supereg